app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt

```diff
@@ -0,0 +1,202 @@
+package com.example.campusguide
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.filters.SdkSuppress
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.*
+import database.ObjectBox
+import org.hamcrest.CoreMatchers.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+private const val TIMEOUT = 5000L
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@SdkSuppress(minSdkVersion = 18)
+class DirectionsSystemTest {
+
+    private lateinit var device: UiDevice
+    private val travelModes = arrayOf("driving", "walking", "transit")
+
+    @get:Rule
+    var permissionRule: GrantPermissionRule? = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @Before
+    fun startMainActivityFromHomeScreen() {
```

Comment:
I think this one was used in several test classes now, should we extract it into a SystemTestUtils class?

Reply:
Good point, done